crates/bevy_ecs/src/event/base.rs

```diff
@@ -11,33 +11,79 @@ use core::{
     marker::PhantomData,
 };
 
```

Comment:
This `base.rs` file is kinda weird because it now has `Event`, `EntityEvent`, and `BufferedEvent`, when it should maybe only have `Event`. But I feel like the modules here need some larger restructuring since literally all the other files in this `event` directory are actually only related to buffered events.

I'd probably restructure these so that `event/base.rs` contains `Event` and related types, `event/entity_event.rs` has `EntityEvent`, and then there'd be an `event/buffered` sub-directory that has everything related to `BufferedEvent`. I didn't want to make such module changes in this PR though, so I thought it'd make more sense to do in a follow-up.

Reply:
Yep, happy to leave that to follow-up.